src/Microsoft.Azure.WebJobs.Host/JobHost.cs

```diff
@@ -23,7 +24,7 @@ namespace Microsoft.Azure.WebJobs
     /// A <see cref="JobHost"/> is the execution container for jobs. Once started, the
     /// <see cref="JobHost"/> will manage and run job functions when they are triggered.
     /// </summary>
-    public class JobHost : IDisposable
+    public class JobHost : IDisposable, IServiceProvider
```

Comment:
Yeah, we definitely shouldn't do this - it commits us to a path that we don't want to go down, and we'll be changing relatively soon anyways when we rework DI on core. We don't want 2 service providers (JobHostConfiguration) is already a SP.